app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlView.kt

```diff
@@ -207,13 +207,10 @@ class SessionControlView(
 
                     JumpBackInCFRDialog(view).showIfNeeded()
 
-                    if (context.settings().showSyncCFR) {
-                        SyncCFRPresenter(
-                            context = context,
-                            recyclerView = view,
-                        ).showSyncCFR()
-                        context.settings().showSyncCFR = false
-                    }
```

Comment:
I am wondering why moving this if check inside of `showSyncCFR` fixes the problem. 

From my point of view, I would like to avoid doing any type of work such as initializing `SyncCFRPresenter` and finding the sync tab view if `showSyncCFR` is already false.

Reply:
I move the `showSyncCFR` check back in SessionControlView, but noticed the same change applied [here](https://github.com/mozilla-mobile/fenix/pull/26757). Should we close this and handle both scenarios in #26757?